examples/renderers/NMR/example4.py

```diff
@@ -0,0 +1,196 @@
+        super(Model, self).__init__()
+
+        ###########################
+        # Load mesh
+        ###########################
+
+        mesh = TriangleMesh.from_obj(mesh_path)
+        # Normalize into unit cube, and expand such that batch size = 1
+        vertices = normalize_vertices(mesh.vertices).unsqueeze(0)
+        faces = mesh.faces.unsqueeze(0)
+
+        self.register_buffer('vertices', vertices)
+        self.register_buffer('faces', faces)
+
+        ###########################
+        # Initialize texture (NMR format)
+        ###########################
+
+        textures = torch.ones(
+            1, self.faces.shape[1], TEXTURE_SIZE, TEXTURE_SIZE, TEXTURE_SIZE,
+            3, dtype=torch.float32
+        )
+        self.register_buffer('textures', textures)
+
+        ###########################
+        # Load target image
+        ###########################
+
+        image_ref = torch.from_numpy(
+            (imread(image_path)[:, :, :3].max(-1) > 0.1).astype(np.float32)
+        )[None, ::]
+        self.register_buffer('image_ref', image_ref)
+
+        from skimage.io import imsave
+        imsave(image_path + '.test.png', image_ref.numpy().transpose((1, 2, 0)))
+
+        ###########################
+        # Initialize camera position
+        ###########################
+
+        self.camera_position = nn.Parameter(
+            torch.from_numpy(np.array(INITIAL_CAMERA_POS, dtype=np.float32)))
+
+        ###########################
+        # Setup renderer
+        ###########################
+
+        renderer = Renderer(camera_mode='look_at')
+        renderer.eye = self.camera_position
+        self.renderer = renderer
+
+    def forward(self):
+        ###########################
+        # Render
+        ###########################
+
+        image = self.renderer(self.vertices, self.faces, mode='silhouettes')
+        loss = torch.sum((image - self.image_ref[None, :, :]) ** 2)
+
+        return loss
+
+
+def main():
+    args = parse_arguments()
+
+    ###########################
+    # Setup model
+    ###########################
+
+    model = Model(args.mesh, args.image)
+    model.cuda()
+
+    ###########################
+    # Optimize
+    ###########################
+
+    loop = tqdm.tqdm(range(args.epochs))
+    loop.set_description('Optimizing')
+
+    optimizer = torch.optim.Adam(
+        filter(lambda p: p.requires_grad, model.parameters()),
+        lr=0.1
+    )
+
+    os.makedirs(args.output_path, exist_ok=True)
+    writer = imageio.get_writer(os.path.join(
+        args.output_path, 'example4.gif'), mode='I')
+    for i in loop:
+        optimizer.zero_grad()
+
+        loss = model()
+
+        loss.backward()
+        optimizer.step()
+
+        images, _, _ = model.renderer(
+            model.vertices,
+            model.faces,
+            model.textures
+        )
+
+        image = images.detach().cpu().numpy()[0].transpose(
```

Comment:
May not be a big deal but you can use `permute` on gpu before transfering to cpu